salt/modules/napalm_bgp.py

```diff
@@ -0,0 +1,282 @@
+# -*- coding: utf-8 -*-
+'''
+NAPALM BGP
+==========
+
+Manages BGP configuration on network devices and provides statistics.
+
+:codeauthor: Mircea Ulinic <mircea@cloudflare.com> & Jerome Fleury <jf@cloudflare.com>
+:maturity:   new
+:depends:    napalm
+:platform:   linux
+
+Dependencies
+------------
+
+- :doc:`napalm proxy minion (salt.proxy.napalm) </ref/proxy/all/salt.proxy.napalm>`
+
+.. versionadded: Carbon
+'''
+
+from __future__ import absolute_import
+
+# Import python lib
+import logging
+log = logging.getLogger(__file__)
+
+
+try:
+    # will try to import NAPALM
+    # https://github.com/napalm-automation/napalm
+    # pylint: disable=W0611
+    from napalm import get_network_driver
+    # pylint: enable=W0611
+    HAS_NAPALM = True
+except ImportError:
+    HAS_NAPALM = False
+
+# ----------------------------------------------------------------------------------------------------------------------
+# module properties
+# ----------------------------------------------------------------------------------------------------------------------
+
+__virtualname__ = 'bgp'
+__proxyenabled__ = ['napalm']
+# uses NAPALM-based proxy to interact with network devices
+
+# ----------------------------------------------------------------------------------------------------------------------
+# property functions
+# ----------------------------------------------------------------------------------------------------------------------
+
+
+def __virtual__():
+
+    '''
+    NAPALM library must be installed for this module to work.
+    Also, the key proxymodule must be set in the __opts___ dictionary.
+    '''
+
+    if HAS_NAPALM and 'proxy' in __opts__:
+        return __virtualname__
+    else:
+        return (False, 'The module napalm_bgp (BGP) cannot be loaded: \
+                napalm lib or proxy could not be loaded.')
+
+# ----------------------------------------------------------------------------------------------------------------------
+# helper functions -- will not be exported
+# ----------------------------------------------------------------------------------------------------------------------
+
+# ----------------------------------------------------------------------------------------------------------------------
+# callable functions
+# ----------------------------------------------------------------------------------------------------------------------
+
+
+def config(group='', neighbor=''):
+
+    '''
+    Provides the BGP configuration on the device.
+
+    :param group: Name of the group selected to display the configuration.
+    :param neighbor: IP Address of the neighbor to display the configuration.
+    The parameter *group* is mandatory to be specified, otherwise *neighbor* will be ignored.
```

Comment:
Ah ok. Maybe that could be reworded to be more clear? Something like "If the group parameter is not specified, the neighbor setting will be ignored." I know the difference is subtle, and maybe it will be more clear once the docs are displayed. Up to you. :)